ui/src/dashboards/reducers/dashTimeV1.ts

```diff
@@ -32,13 +35,29 @@ export default (state: State = initialState, action: Action) => {
       return {...state, ranges}
     }
 
+    case ActionType.RetainDashboardRefresh: {
+      const {dashboardIDs} = action.payload
+      const refreshes = state.refreshes.filter(r =>
+        dashboardIDs.includes(r.dashboardID)
```

Comment:
might be worth keeping an eye on this action as it runs in `O(mn)` time (looping over `refreshes` then looping over `dashboardIds`), which [I guess is considered multilinear](https://cs.stackexchange.com/questions/9523/is-omn-considered-linear-or-quadratic-growth). tldr: a ton of a dashboards _might_ make this slow... 🤔

Reply:
Interesting observation. This issue also applies to `ActionType.RetainRangesDashboardTimeV1`.

Instead of keep it as an array, I propose using a hash for the `dashboardIDs`, thus allowing O(1) lookup time (rather than O(n) using `.includes`) inside the filter callback.